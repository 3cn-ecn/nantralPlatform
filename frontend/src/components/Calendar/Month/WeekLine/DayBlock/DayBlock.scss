.imageMonthEventBlur {
  width: inherit;
  height: inherit;
  backdrop-filter: blur(3px) brightness(65%);
  display: flex;
  justify-content: center;
  align-items: center;
  border-radius: inherit;
}

.dayBlockTitleDisplay {
  overflow: hidden;
  overflow-wrap: break-word;
  display: -webkit-box;
   -webkit-line-clamp: 2; // Allow the title on 2 lines maximum
   -webkit-box-orient: vertical;
}