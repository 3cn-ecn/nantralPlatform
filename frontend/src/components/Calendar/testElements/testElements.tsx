import { EventProps } from '../../../Props/Event';

export function createTestEvent() {
  const event: EventProps = {
    color: 'blue',
    beginDate: new Date('2023-02-17T03:24:00'),
    endDate: new Date('2023-02-17T04:24:00'),
    description: 'This is an event to test',
    getAbsoluteUrl: 'https://nantral-platform.fr/account/login/?next=/',
    getGroupName: 'Moi-mÃªme',
    group: 'Mon groupe',
    image: null,
    isMember: true,
    isParticipating: true,
    location: 'My home',
    numberOfParticipants: 15,
    publicity: 'Pub',
    slug: 'The slug',
    title: 'The test event',
    maxParticipant: null,
    endInscription: null,
    beginInscription: null,
    ticketing: null,
    isFavorite: true,
    effectiveSize: 1,
    globalSize: 1,
    position: 0,
    placed: false,
  };

  return event;
}
