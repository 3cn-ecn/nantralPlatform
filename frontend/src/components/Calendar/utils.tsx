/**
 * Get the number of days in a month.
 * @param year The year of the month.
 * @param month The month.
 * @returns The number of days in the month.
 */
export function numberOfDayInMonth(year: number, month: number): number {
  return new Date(year, month + 1, 0).getDate();
}

/**
 * Get the number of days in a month.
 * @param date The date in the month.
 * @returns The number of days in the month.
 */
export function numberOfDayInDateMonth(date: Date): number {
  return new Date(date.getFullYear(), date.getMonth() + 1, 0).getDate();
}

/**
 * Get first day of a month
 * @param date A date in the month.
 * @returns The first day of the month.
 */
export function firstMonthDay(date: Date): Date {
  return new Date(date.getFullYear(), date.getMonth(), 0);
}
