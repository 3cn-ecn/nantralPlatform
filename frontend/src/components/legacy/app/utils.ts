/**
 * function to hash a string
 * @param base64String A string not hashed
 * @returns A hashed string
 */
export function urlBase64ToUint8Array(base64String) {
  var padding = '='.repeat((4 - (base64String.length % 4)) % 4);
  var base64 = (base64String + padding).replace(/\-/g, '+').replace(/_/g, '/');
  var rawData = window.atob(base64);
  var outputArray = new Uint8Array(rawData.length);
  for (var i = 0; i < rawData.length; ++i) {
    outputArray[i] = rawData.charCodeAt(i);
  }
  return outputArray;
}

/**
 * Load the version of browser
 * @returns A dict with name and version of the browser
 */
export function loadVersionBrowser() {
  if ('userAgentData' in navigator) {
    // navigator.userAgentData is not available in
    // Firefox and Safari
    const uaData = (navigator as any).userAgentData;
    // Outputs of navigator.userAgentData.brands[n].brand are e.g.
    // Chrome: 'Google Chrome'
    // Edge: 'Microsoft Edge'
    // Opera: 'Opera'
    let browsername;
    let browserversion;
    let chromeVersion = null;
    for (var i = 0; i < uaData.brands.length; i++) {
      let brand = uaData.brands[i].brand;
      browserversion = uaData.brands[i].version;
      if (
        brand.match(/opera|chrome|edge|safari|firefox|msie|trident/i) !== null
      ) {
        // If we have a chrome match, save the match, but try to find another match
        // E.g. Edge can also produce a false Chrome match.
        if (brand.match(/chrome/i) !== null) {
          chromeVersion = browserversion;
        }
        // If this is not a chrome match return immediately
        else {
          browsername = brand.substr(brand.indexOf(' ') + 1);
          return {
            name: browsername,
            version: browserversion,
          };
        }
      }
    }
    // No non-Chrome match was found. If we have a chrome match, return it.
    if (chromeVersion !== null) {
      return {
        name: 'chrome',
        version: chromeVersion,
      };
    }
  }
  // If no userAgentData is not present, or if no match via userAgentData was found,
  // try to extract the browser name and version from userAgent
  const userAgent = navigator.userAgent;
  var ua = userAgent,
    tem,
    M =
      ua.match(
        /(opera|chrome|safari|firefox|msie|trident(?=\/))\/?\s*(\d+)/i
      ) || [];
  if (/trident/i.test(M[1])) {
    tem = /\brv[ :]+(\d+)/g.exec(ua) || [];
    return { name: 'IE', version: tem[1] || '' };
  }
  if (M[1] === 'Chrome') {
    tem = ua.match(/\bOPR\/(\d+)/);
    if (tem != null) {
      return { name: 'Opera', version: tem[1] };
    }
  }
  M = M[2] ? [M[1], M[2]] : [navigator.appName, navigator.appVersion, '-?'];
  if ((tem = ua.match(/version\/(\d+)/i)) != null) {
    M.splice(1, 1, tem[1]);
  }
  return {
    name: M[0],
    version: M[1],
  };
}
