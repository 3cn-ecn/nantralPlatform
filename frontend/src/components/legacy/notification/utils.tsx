/**
 * Merge two lists of objects
 * @param a First list
 * @param b Second list
 * @returns The merged list
 */
export default function merge(a:any[], b:any[]) {
    if (a == null) a = [];
    if (b == null) b = [];
    return [...new Set([...a, ...b])];
}