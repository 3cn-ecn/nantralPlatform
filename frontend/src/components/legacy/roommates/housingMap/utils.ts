import axios from "../../utils/axios";
import { Housing } from "./interfaces";

export async function getRoommates(
  api_housing_url: string,
  setColocs,
  setData
): Promise<void> {
  axios
    .get(api_housing_url)
    .then((res) => {
      // For some reason, doubles Axios roommates which have more than one inhabitant,
      // so we have to do this mess to filter everything.
      // Hours wasted: 2
      var uniqueIds: number[] = [];
      let dataBuffer = res.data.filter((e, i) => {
        if (!uniqueIds.includes(e.id)) {
          uniqueIds.push(e.id);
          return true;
        }
        return false;
      });
      // Here, we avoid problems with roommates in the same building
      // (on different floors for example)
      // If two roommates are on the same coordinates, we shift them slightly so they don't overlap.
      let i = 0,
        j = 0;
      for (i = 0; i < dataBuffer.length; i++) {
        for (j = i + 1; j < dataBuffer.length; j++) {
          if (dataBuffer[i].longitude === dataBuffer[j].longitude) {
            dataBuffer[j].longitude += 0.00001;
          }
        }
      }
      setColocs(
        dataBuffer.map((housing: Housing) => {
          housing.address = housing.address.replace(
            ", 44100 Nantes, France",
            ""
          );
          return { label: housing.roommates.name, housing: housing };
        })
      );
      setData(dataBuffer);
    })
    .catch((err) => {
      setData([]);
    });
}
