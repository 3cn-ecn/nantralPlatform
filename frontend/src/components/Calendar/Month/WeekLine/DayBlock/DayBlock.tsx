import React from 'react';
import { useNavigate } from 'react-router-dom';

import { Box, Button } from '@mui/material';

import { EventProps } from '#types/Event';

import './DayBlock.scss';

export function DayBlock(props: {
  day: { day: number; date: number; events: Array<EventProps> } | null;
  inMonth: boolean;
  maxEventsInDayWeek: number;
}): JSX.Element {
  const { day, inMonth, maxEventsInDayWeek } = props;
  const navigate = useNavigate();
  if (inMonth) {
    return (
      <Box
        sx={{
          height: `${(maxEventsInDayWeek + 0.6) * 3}rem`,
          border: '1px solid gray',
        }}
        data-testid="DateBoxTestId"
      >
        {day.date}
        {day.events.map((event) => {
          return (
            <Button
              key={event.slug}
              variant="contained"
              fullWidth
              onClick={() => {
                navigate(`/event/${event.id}/`);
              }}
              style={{
                minWidth: `1px`,
                height: `3rem`,
                padding: '0px',
                background: `url(${event.image})`,
                backgroundSize: 'cover',
                backgroundPosition: 'center',
                backgroundRepeat: 'no-repeat',
                textTransform: 'none',
              }}
              data-testid={`${event.slug}DateBoxButtonTestId`}
            >
              <div className="imageMonthEventBlur">
                <div className="dayBlockTitleDisplay">{event.title}</div>
              </div>
            </Button>
          );
        })}
      </Box>
    );
  }
  return <div></div>;
}
