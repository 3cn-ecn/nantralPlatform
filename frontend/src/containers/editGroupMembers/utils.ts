import axios from "axios";

import { Member } from "../groupMembers/interfaces";
import { getMembers } from "../groupMembers/utils";

export function membersSort(a: Member, b: Member): number {
  if (a.order === 0) {
    return -1;
  }
  if (b.order === 0) {
    return -1;
  }
  return a.order - b.order;
}

export function updateMember(
  membersUrl: string,
  member: Member,
  role: string,
  beginDate: string,
  endDate: string,
  admin: boolean,
  handleClose: () => void,
  setIsUpdateLoading: React.Dispatch<React.SetStateAction<boolean>>,
  setIsLoading: React.Dispatch<React.SetStateAction<boolean>>,
  setMembers: React.Dispatch<React.SetStateAction<Member[]>>
): void {
  setIsUpdateLoading(true);
  axios
    .post(membersUrl, {
      editMode: 2,
      id: member.id,
      role: role,
      beginDate: beginDate,
      endDate: endDate,
      admin: admin,
    })
    .then((resp) => {
      if (resp.status !== 200) {
        console.error(
          "Le membre n'a pas pu être mis à jour.",
          "ERR:",
          resp.status
        );
      }
    })
    .catch((e) => {
      console.error("Le membre n'a pas pu être mis à jour.");
    })
    .finally(() => {
      setIsUpdateLoading(false);
      handleClose();
      getMembers(membersUrl, setMembers, setIsLoading);
    });
}

export function deleteMember(
  membersUrl: string,
  member: Member,
  handleClose: () => void,
  setIsUpdateLoading: React.Dispatch<React.SetStateAction<boolean>>,
  setIsLoading: React.Dispatch<React.SetStateAction<boolean>>,
  setMembers: React.Dispatch<React.SetStateAction<Member[]>>
): void {
  setIsUpdateLoading(true);
  axios
    .post(membersUrl, {
      editMode: 3,
      id: member.id,
    })
    .then((resp) => {
      if (resp.status !== 200) {
        console.error(
          "Le membre n'a pas pu être supprimé.",
          "ERR:",
          resp.status
        );
      }
    })
    .catch((e) => {
      console.error("Le membre n'a pas pu être supprimé.");
    })
    .finally(() => {
      setIsUpdateLoading(false);
      handleClose();
      getMembers(membersUrl, setMembers, setIsLoading);
    });
}
