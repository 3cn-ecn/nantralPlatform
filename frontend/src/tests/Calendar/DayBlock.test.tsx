import React from 'react';
import { BrowserRouter } from 'react-router-dom';

import { DayBlock } from '#shared/components/Calendar/Month/WeekLine/DayBlock/DayBlock';
import { wrapAndRender } from '#shared/utils/tests';
import { EventProps } from '#types/Event';

import { createTestEvent } from './testElements/testElements';

const events: Array<EventProps> = [];
events.push(
  createTestEvent(
    'The slug',
    'The test event',
    new Date('2023-02-17T03:24:00'),
    new Date('2023-02-17T04:24:00')
  )
);
events.push(
  createTestEvent(
    'The new slug',
    'Another test event',
    new Date('2023-02-17T08:24:00'),
    new Date('2023-02-17T10:24:00')
  )
);
const day = { day: 5, date: 17, events: events };

const otherEvents: Array<EventProps> = [];
otherEvents.push(
  createTestEvent(
    undefined,
    undefined,
    new Date('2023-02-17T03:24:00'),
    new Date('2023-02-17T04:24:00')
  )
);
const otherDay = { day: 5, date: 17, events: otherEvents };

const newOtherDay = { day: 5, date: 17, events: [] };
describe('<DayBlock />', () => {
  it('should display a DayBlock', async () => {
    const component = wrapAndRender(
      <BrowserRouter>
        <DayBlock
          key="DayBlockTest1"
          day={day}
          maxEventsInDayWeek={2}
          inMonth
        />
      </BrowserRouter>
    );
    let innerHTMLString = '17';

    day.events.forEach((event) => {
      expect(
        component.getByTestId(`${event.slug}DateBoxButtonTestId`).style.height
      ).toBe('3rem');
      expect(
        component.getByTestId(`${event.slug}DateBoxButtonTestId`).textContent
      ).toBe(event.title);
      innerHTMLString += event.title;
    });
    expect(component.getByTestId('DateBoxTestId').textContent).toBe(
      innerHTMLString
    );

    expect(component).toMatchSnapshot();
  });

  it('should display an unfullfilled DayBlock', async () => {
    const component = wrapAndRender(
      <BrowserRouter>
        <DayBlock
          key="DayBlockTest1"
          day={otherDay}
          maxEventsInDayWeek={3}
          inMonth
        />
      </BrowserRouter>
    );
    let innerHTMLString = '17';

    otherDay.events.forEach((event) => {
      expect(
        component.getByTestId(`${event.slug}DateBoxButtonTestId`).style.height
      ).toBe('3rem');
      expect(
        component.getByTestId(`${event.slug}DateBoxButtonTestId`).textContent
      ).toBe(event.title);
      innerHTMLString += event.title;
    });
    expect(component.getByTestId('DateBoxTestId').textContent).toBe(
      innerHTMLString
    );

    expect(component).toMatchSnapshot();
  });

  it('should display an empty DayBlock', async () => {
    const component = wrapAndRender(
      <BrowserRouter>
        <DayBlock
          key="DayBlockTest1"
          day={newOtherDay}
          maxEventsInDayWeek={5}
          inMonth
        />
      </BrowserRouter>
    );
    expect(component.getByTestId('DateBoxTestId').textContent).toBe('17');

    expect(component).toMatchSnapshot();
  });

  it('should display an out of month DayBlock', async () => {
    const component = wrapAndRender(
      <BrowserRouter>
        <DayBlock
          key="DayBlockTest4"
          day={null}
          maxEventsInDayWeek={1}
          inMonth={false}
        />
      </BrowserRouter>
    );
    expect(component).toMatchSnapshot();
  });
});
