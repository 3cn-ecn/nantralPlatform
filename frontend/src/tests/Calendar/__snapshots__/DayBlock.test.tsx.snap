// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DayBlock /> should display a DayBlock 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiBox-root css-1e8h3rg"
        data-testid="DateBoxTestId"
      >
        17
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-fullWidth MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-fullWidth css-1fu7jd5-MuiButtonBase-root-MuiButton-root"
          data-testid="The slugDateBoxButtonTestId"
          style="min-width: 1px; height: 3rem; padding: 0px; background: url(null); background-size: cover; background-position: center; background-repeat: no-repeat; text-transform: none;"
          tabindex="0"
          type="button"
        >
          <div
            class="imageMonthEventBlur"
          >
            <div
              class="dayBlockTitleDisplay"
            >
              The test event
            </div>
          </div>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-fullWidth MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-fullWidth css-1fu7jd5-MuiButtonBase-root-MuiButton-root"
          data-testid="The new slugDateBoxButtonTestId"
          style="min-width: 1px; height: 3rem; padding: 0px; background: url(null); background-size: cover; background-position: center; background-repeat: no-repeat; text-transform: none;"
          tabindex="0"
          type="button"
        >
          <div
            class="imageMonthEventBlur"
          >
            <div
              class="dayBlockTitleDisplay"
            >
              Another test event
            </div>
          </div>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiBox-root css-1e8h3rg"
      data-testid="DateBoxTestId"
    >
      17
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-fullWidth MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-fullWidth css-1fu7jd5-MuiButtonBase-root-MuiButton-root"
        data-testid="The slugDateBoxButtonTestId"
        style="min-width: 1px; height: 3rem; padding: 0px; background: url(null); background-size: cover; background-position: center; background-repeat: no-repeat; text-transform: none;"
        tabindex="0"
        type="button"
      >
        <div
          class="imageMonthEventBlur"
        >
          <div
            class="dayBlockTitleDisplay"
          >
            The test event
          </div>
        </div>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-fullWidth MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-fullWidth css-1fu7jd5-MuiButtonBase-root-MuiButton-root"
        data-testid="The new slugDateBoxButtonTestId"
        style="min-width: 1px; height: 3rem; padding: 0px; background: url(null); background-size: cover; background-position: center; background-repeat: no-repeat; text-transform: none;"
        tabindex="0"
        type="button"
      >
        <div
          class="imageMonthEventBlur"
        >
          <div
            class="dayBlockTitleDisplay"
          >
            Another test event
          </div>
        </div>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`<DayBlock /> should display an empty DayBlock 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiBox-root css-8hisd4"
        data-testid="DateBoxTestId"
      >
        17
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiBox-root css-8hisd4"
      data-testid="DateBoxTestId"
    >
      17
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`<DayBlock /> should display an out of month DayBlock 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div />
    </div>
  </body>,
  "container": <div>
    <div />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`<DayBlock /> should display an unfullfilled DayBlock 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiBox-root css-1vxav2g"
        data-testid="DateBoxTestId"
      >
        17
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-fullWidth MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-fullWidth css-1fu7jd5-MuiButtonBase-root-MuiButton-root"
          data-testid="The slugDateBoxButtonTestId"
          style="min-width: 1px; height: 3rem; padding: 0px; background: url(null); background-size: cover; background-position: center; background-repeat: no-repeat; text-transform: none;"
          tabindex="0"
          type="button"
        >
          <div
            class="imageMonthEventBlur"
          >
            <div
              class="dayBlockTitleDisplay"
            >
              The test event
            </div>
          </div>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiBox-root css-1vxav2g"
      data-testid="DateBoxTestId"
    >
      17
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-fullWidth MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-fullWidth css-1fu7jd5-MuiButtonBase-root-MuiButton-root"
        data-testid="The slugDateBoxButtonTestId"
        style="min-width: 1px; height: 3rem; padding: 0px; background: url(null); background-size: cover; background-position: center; background-repeat: no-repeat; text-transform: none;"
        tabindex="0"
        type="button"
      >
        <div
          class="imageMonthEventBlur"
        >
          <div
            class="dayBlockTitleDisplay"
          >
            The test event
          </div>
        </div>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
